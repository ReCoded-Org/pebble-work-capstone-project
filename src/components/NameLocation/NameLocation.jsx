import Image from "next/image";


import Button from "@/components/Button";

import InputComponent from "../InputComponent";

const NameLocation = () => {
    return (
        <div className='m-4 mt-4 md:m-12 lg:ml-16'>
            <h1 className='m-4 text-xl font-medium md:m-6 md:text-3xl lg:m-12 lg:text-5xl'>
                Edit Profile
            </h1>
            <div className='flex flex-row items-center justify-center overflow-hidden lg:ml-8 lg:justify-start'>
                <div className='m-1 mr-2 w-20 overflow-hidden md:mr-6 md:w-40 lg:w-44'>
                    <Image
                        src='/images/user.png'
                        alt='name-location'
                        width='100%'
                        height='100%'
                        layout='responsive'
                    />
                </div>
                <div className='m-1 md:m-2'>
                    <Button
                        label='Upload New'
                        bgColor='bg-primary-200'
                        textColor='text-white'
                        fontWeight='font-medium'
                        width='w-20 md:w-44 lg:w-48'
                        height='h-8 md:h-14 lg:h-14'
                        borderColor=''
                        fontSize='text-[11px] md:text-xl lg:text-xl'
                        border=''
                    />
                </div>
                <div className='m-1 md:m-2'>
                    <Button
                        label='Choose From Library'
                        bgColor='bg-white'
                        textColor='text-black'
                        fontWeight='font-medium'
                        width='w-30 md:w-60 lg:w-64'
                        height='h-8 md:h-14 lg:h-14'
                        borderColor='border-black'
                        fontSize='text-[11px] md:text-xl lg:text-xl'
                        border='border border-b-2 border-r-2'
                    />
                </div>
            </div>
            <div className='m-4 mt-8 md:m-8 md:mt-12'>
                <label className='text-[12px] font-medium md:text-xl lg:text-xl'>
                    Name (Required)
                </label>
                <div className='mt-1 mb-3 h-8 md:mt-2 md:mb-9 md:h-14 md:w-[490px] lg:h-12  lg:w-[600px]'>
                    <input
                        type='text'
                        name='Name'
                        placeholder='Name'
                        required
                        className='h-full w-full rounded-md border-2 border-black p-2'
                    />
                </div>
                <label className='text-[12px] font-medium md:text-xl lg:text-xl'>
                    Your Location
                </label>
                <div className='mt-1 mb-3 h-8 md:mt-2 md:mb-9 md:h-14 md:w-[490px] lg:h-12 lg:w-[600px]'>
                    <InputComponent placeholder='Location' />
                </div>
            </div>
        </div>
    );
};

export default NameLocation;
