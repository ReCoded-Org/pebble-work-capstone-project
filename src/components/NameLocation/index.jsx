import NameLocation from "./NameLocation";
export default NameLocation;
