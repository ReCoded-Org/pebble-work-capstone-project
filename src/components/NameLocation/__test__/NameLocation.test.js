import renderer from "react-test-renderer";

import NameLocation from "../NameLocation";

it("renders correctly", () => {
    const tree = renderer.create(<NameLocation />).toJSON();
    expect(tree).toMatchSnapshot();
});
