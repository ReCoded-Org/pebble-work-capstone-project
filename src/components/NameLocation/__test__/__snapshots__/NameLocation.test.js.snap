// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="m-4 mt-4 md:m-12 lg:ml-16"
>
  <h1
    className="m-4 text-xl font-medium md:m-6 md:text-3xl lg:m-12 lg:text-5xl"
  >
    Edit Profile
  </h1>
  <div
    className="flex flex-row items-center justify-center overflow-hidden lg:ml-8 lg:justify-start"
  >
    <div
      className="m-1 mr-2 w-20 overflow-hidden rounded-full md:mr-6 md:w-40 lg:w-44"
    >
      <span
        style={
          Object {
            "background": "none",
            "border": 0,
            "boxSizing": "border-box",
            "display": "block",
            "height": "initial",
            "margin": 0,
            "opacity": 1,
            "overflow": "hidden",
            "padding": 0,
            "position": "relative",
            "width": "initial",
          }
        }
      >
        <span
          style={
            Object {
              "background": "none",
              "border": 0,
              "boxSizing": "border-box",
              "display": "block",
              "height": "initial",
              "margin": 0,
              "opacity": 1,
              "padding": 0,
              "paddingTop": "100%",
              "width": "initial",
            }
          }
        />
        <img
          alt="name-location"
          data-nimg="responsive"
          decoding="async"
          onError={[Function]}
          onLoad={[Function]}
          src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
          style={
            Object {
              "border": "none",
              "bottom": 0,
              "boxSizing": "border-box",
              "display": "block",
              "height": 0,
              "left": 0,
              "margin": "auto",
              "maxHeight": "100%",
              "maxWidth": "100%",
              "minHeight": "100%",
              "minWidth": "100%",
              "objectFit": "cover",
              "objectPosition": undefined,
              "padding": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
              "width": 0,
            }
          }
        />
        <noscript>
          <img
            alt="name-location"
            data-nimg="responsive"
            decoding="async"
            loading="lazy"
            sizes="100vw"
            src="/_next/image?url=%2Fimages%2Fuser.png&w=3840&q=75"
            srcSet="/_next/image?url=%2Fimages%2Fuser.png&w=640&q=75 640w, /_next/image?url=%2Fimages%2Fuser.png&w=750&q=75 750w, /_next/image?url=%2Fimages%2Fuser.png&w=828&q=75 828w, /_next/image?url=%2Fimages%2Fuser.png&w=1080&q=75 1080w, /_next/image?url=%2Fimages%2Fuser.png&w=1200&q=75 1200w, /_next/image?url=%2Fimages%2Fuser.png&w=1920&q=75 1920w, /_next/image?url=%2Fimages%2Fuser.png&w=2048&q=75 2048w, /_next/image?url=%2Fimages%2Fuser.png&w=3840&q=75 3840w"
            style={
              Object {
                "border": "none",
                "bottom": 0,
                "boxSizing": "border-box",
                "display": "block",
                "height": 0,
                "left": 0,
                "margin": "auto",
                "maxHeight": "100%",
                "maxWidth": "100%",
                "minHeight": "100%",
                "minWidth": "100%",
                "objectFit": "cover",
                "objectPosition": undefined,
                "padding": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
                "width": 0,
              }
            }
          />
        </noscript>
      </span>
    </div>
    <div
      className="m-1 flex flex-col md:m-2"
    >
      <input
        accept="image/x-png,image/gif,image/jpeg"
        onChange={[Function]}
        type="file"
      />
    </div>
  </div>
  <div
    className="m-4 mt-8 md:m-8 md:mt-12"
  >
    <label
      className="text-[12px] font-medium md:text-xl lg:text-xl"
    >
      Name (Required)
    </label>
    <div
      className="mt-1 mb-3 h-8 md:mt-2 md:mb-9 md:h-14 md:w-[490px] lg:h-12 lg:w-[600px]"
    >
      <input
        className="h-full w-full rounded-md border-2 border-black p-2"
        name="Name"
        onChange={[Function]}
        placeholder="Name"
        required={true}
        type="text"
      />
    </div>
    <label
      className="text-[12px] font-medium md:text-xl lg:text-xl"
    >
      Preferred cities:
    </label>
    <div
      className="mt-1 mb-3 h-8 md:mt-2 md:mb-9 md:h-14 md:w-[490px] lg:h-12 lg:w-[600px]"
    >
      <div
        className="mt-1 mb-3 h-8 md:mt-2 md:mb-9 md:h-14 md:w-[490px] lg:h-12 lg:w-[600px]"
      >
        <input
          className="h-full w-full rounded-md border-2 border-black p-2"
          name="Location"
          onClick={[Function]}
          placeholder="Location"
          required={true}
          type="text"
          value={Array []}
        />
      </div>
    </div>
  </div>
  <div
    className="flex justify-end pr-8"
  >
    <button
      className="mt-1 mb-1 w-20 md:w-32 lg:w-36 h-8 md:h-12 lg:h-12 text-[11px] md:text-xl lg:text-xl bg-primary-200  undefined hover:undefined text-white font-medium  py-1  px-2  hover:undefined rounded  hover:shadow-lg undefined"
      onClick={[Function]}
      type="button"
    >
      Submit
    </button>
  </div>
</div>
`;
